src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -549,7 +551,22 @@ public static IDisposable BuildEnhancedLinePragma(this CodeWriter writer, Source
             return NullDisposable.Default;
         }
 
-        return new LinePragmaWriter(writer, span.Value, context, characterOffset, useEnhancedLinePragma: true);
+        var sourceSpan = RemapFilePathIfNecessary(span.Value, context);
+        return new LinePragmaWriter(writer, sourceSpan, context, characterOffset, useEnhancedLinePragma: true);
+    }
+
+    private static SourceSpan RemapFilePathIfNecessary(SourceSpan sourceSpan, CodeRenderingContext context)
+    {
+        if (context.Options.RemapLinePragmaPathsOnWindows && RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Should we add a `&& sourceSpan.FilePath.Contains('/')` here so we avoid the `Replace` work if it's not necessary?

Reply:
We'd need to benchmark it but honestly that might be slower. We almost always expect these to be re-mapped on windows, so will usually have to do the replace. Looking at replace source code it basically does a `Contains` to find the indices to replace, before actually replacing them. That bails early if it doesn't find anything, so in the case where there are no chars to replace, checking first is the same speed, and when there are replacements its slower.